# HG changeset patch
# User Rik <rik@octave.org>
# Date 1423101298 28800
#      Wed Feb 04 17:54:58 2015 -0800
# Node ID 40e846a96908fc5853940fc0f70a035c6c72aa48
# Parent  a9516bc4c55c161047d0910ca7f72252a9163c1f
graphics_toolkit.m: Return a valid toolkit, not "", before any plotting has happened (bug #41665).

* graphics_toolkit.m: Check return value of graphics_toolkit.  If empty, look
for 'qt', then 'fltk', then any available plotting engine.

diff --git a/scripts/plot/util/graphics_toolkit.m b/scripts/plot/util/graphics_toolkit.m
--- a/scripts/plot/util/graphics_toolkit.m
+++ b/scripts/plot/util/graphics_toolkit.m
@@ -39,16 +39,27 @@
 function retval = graphics_toolkit (name, hlist = [])
 
   if (nargin > 2)
     print_usage ();
   endif
 
   if (nargout > 0 || nargin == 0)
     retval = get (0, "defaultfigure__graphics_toolkit__");
+    ## Handle case where graphics_toolkit has been called before any plotting
+    if (isempty (retval))
+      toolkits = available_graphics_toolkits ();
+      if (any (strcmp ("qt", toolkits)))
+        retval = "qt";
+      elseif (any (strcmp ("fltk", toolkits)))
+        retval = "fltk";
+      elseif (! isempty (toolkits))
+        retval = toolkits{1};
+      endif
+    endif
   endif
 
   if (nargin == 0)
     return;
   elseif (nargin == 1)
     if (all (isfigure (name)))
       hlist = name;
       retval = get (hlist, "__graphics_toolkit__");
